src/System.Collections/tests/Generic/List/ForEach_Action.cs

```diff
@@ -107,6 +107,8 @@ public void CallForEach(T[] items, Action<T> action)
             for (int i = 0; i < items.Length; ++i)
                 list.Add(items[i]);
 
+            Assert.Throws<ArgumentNullException>("action", () => list.ForEach(null));
+
```

Comment:
This won't pass.

It's a bit confusing, but the ```List<T>``` in corefx isn't actually used when running on CoreCLR, only with .NET Native.  When System.Collections.dll gets built for CoreCLR, it doesn't actually contain this ```List<T>``` implementation but rather type-forwards to the implementation in mscorlib.dll, which is in the coreclr repo.  As such, to affect that implementation, this would need to be changed:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/Generic/List.cs#L542

Fixing the parameter name in the exception is I think a fine thing to do, but really the implementation in CoreCLR should be fixed first.  Then once it's updated and can be consumed from the corefx repo via the NuGet packages, this test can be updated.  At that point the corefx ```List<T>``` can also be updated.  In other words, this PR should be delayed/closed until a corresponding one is made in CoreCLR, and then this can be revisited.